From d09d03388c6c511834d48dfd84874bf077d01f82 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Tue, 29 Mar 2016 14:25:08 +0500
Subject: [PATCH 2/2] allow building gl backends only

For this specific package we'd like to only
build/deploy the gl (EGL, GLX) backends as
this package will mainly take care of the circular
dependencies between mesa and libva.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7c88751..8dde9ca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -289,8 +289,8 @@ pkgconfigdir=${libdir}/pkgconfig
 AC_SUBST(pkgconfigdir)
 
 # Check for builds without backend
-if test "$USE_DRM:$USE_X11:$USE_WAYLAND" = "no:no:no"; then
-    AC_MSG_ERROR([Please select at least one backend (DRM, X11, Wayland)])
+if test "$USE_DRM:$USE_X11:$USE_WAYLAND:$USE_EGL:$USE_GLX" = "no:no:no:no:no"; then
+    AC_MSG_ERROR([Please select at least one backend (DRM, X11, Wayland, EGL, GLX)])
 fi
 
 # Define __vaDriverInit*() function name for dummy backend
diff --git a/pkgconfig/Makefile.am b/pkgconfig/Makefile.am
index a3435a4..4ce8aa1 100644
--- a/pkgconfig/Makefile.am
+++ b/pkgconfig/Makefile.am
@@ -20,8 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-pcfiles		 = libva.pc
-pcfiles		+= libva-tpi.pc
+pcfiles =
 if USE_DRM
 pcfiles		+= libva-drm.pc
 endif
diff --git a/va/Makefile.am b/va/Makefile.am
index 34ee961..1a35c22 100644
--- a/va/Makefile.am
+++ b/va/Makefile.am
@@ -36,26 +36,7 @@ libva_source_c = \
 	va_trace.c		\
 	$(NULL)
 
-libva_source_h = \
-	va.h			\
-	va_backend.h		\
-	va_backend_tpi.h	\
-	va_backend_vpp.h	\
-	va_compat.h		\
-	va_dec_hevc.h		\
-	va_dec_jpeg.h		\
-	va_dec_vp8.h		\
-	va_dec_vp9.h		\
-	va_drmcommon.h		\
-	va_enc_hevc.h		\
-	va_enc_h264.h		\
-	va_enc_jpeg.h		\
-	va_enc_vp8.h		\
-	va_enc_mpeg2.h		\
-	va_tpi.h		\
-	va_version.h		\
-	va_vpp.h		\
-	$(NULL)
+libva_source_h =
 
 libva_source_h_priv = \
 	sysdeps.h		\
@@ -68,20 +49,10 @@ libva_ldflags = \
 	-Wl,-version-script,${srcdir}/libva.syms \
 	$(NULL)
 
-lib_LTLIBRARIES			= libva.la
+lib_LTLIBRARIES	=
 libvaincludedir			= ${includedir}/va
 libvainclude_HEADERS		= $(libva_source_h)
 noinst_HEADERS			= $(libva_source_h_priv)
-libva_la_SOURCES		= $(libva_source_c)
-libva_la_LDFLAGS		= $(libva_ldflags)
-libva_la_DEPENDENCIES		= libva.syms
-libva_la_LIBADD			= $(LIBVA_LIBS) -ldl
-
-lib_LTLIBRARIES			+= libva-tpi.la
-libva_tpi_la_SOURCES		= va_tpi.c
-libva_tpi_la_LDFLAGS		= $(LDADD) -no-undefined
-libva_tpi_la_DEPENDENCIES	= libva.la 
-libva_tpi_la_LIBADD		= libva.la -ldl
 
 if USE_DRM
 SUBDIRS				+= drm
@@ -109,8 +80,8 @@ SUBDIRS				+= glx
 lib_LTLIBRARIES			+= libva-glx.la
 libva_glx_la_SOURCES		=
 libva_glx_la_LDFLAGS		= $(LDADD)
-libva_glx_la_DEPENDENCIES	= libva.la glx/libva_glx.la libva-x11.la
-libva_glx_la_LIBADD		= libva.la glx/libva_glx.la libva-x11.la \
+libva_glx_la_DEPENDENCIES	= glx/libva_glx.la
+libva_glx_la_LIBADD		= glx/libva_glx.la \
 	$(GLX_LIBS) -ldl
 endif
 
@@ -119,8 +90,8 @@ SUBDIRS				+= egl
 lib_LTLIBRARIES			+= libva-egl.la
 libva_egl_la_SOURCES		=
 libva_egl_la_LDFLAGS		= $(LDADD)
-libva_egl_la_DEPENDENCIES	= libva.la egl/libva_egl.la 
-libva_egl_la_LIBADD		= libva.la egl/libva_egl.la \
+libva_egl_la_DEPENDENCIES	= egl/libva_egl.la 
+libva_egl_la_LIBADD		= egl/libva_egl.la \
 	$(EGL_LIBS) -ldl
 endif
 
-- 
1.9.1

